templates/repo/issue/view_content/pull.tmpl

```diff
@@ -427,10 +444,45 @@
 										{{.CsrfTokenHtml}}
 										<input type="hidden" name="head_commit_id" value="{{.PullHeadCommitID}}">
 										<div class="field">
-											<input type="text" name="merge_commit_id"  placeholder="{{$.i18n.Tr "repo.pulls.merge_commit_id"}}">
+											<input type="text" name="merge_title_field" value="{{.Issue.PullRequest.GetDefaultMergeMessage}}">
 										</div>
-										<button class="ui red button" type="submit" name="do" value="manually-merged">
-											{{$.i18n.Tr "repo.pulls.merge_manually"}}
+										<div class="field">
+											<textarea name="merge_message_field" rows="5" placeholder="{{$.i18n.Tr "repo.editor.commit_message_desc"}}">Reviewed-on: {{$.Issue.HTMLURL}}&#13;&#10;{{$approvers}}</textarea>
+										</div>
+										<button class="ui red button" type="submit" name="do" value="merge">
+											{{$.i18n.Tr "repo.pulls.merge_pull_request_now"}}
+										</button>
+										<button class="ui button merge-cancel">
+											{{$.i18n.Tr "cancel"}}
+										</button>
+										{{if .IsPullBranchDeletable}}
+											<div class="ui checkbox ml-2">
+												<input name="delete_branch_after_merge" type="checkbox" {{if $prUnit.PullRequestsConfig.DefaultDeleteBranchAfterMerge}}checked{{end}}>
+												<label>{{$.i18n.Tr "repo.branch.delete" .HeadTarget}}</label>
+											</div>
+										{{end}}
+									</form>
+								</div>
+								<div class="ui form merge-fields" style="display: none">
+									<form action="{{.Link}}/merge" method="post">
+										{{.CsrfTokenHtml}}
+										{{if and $statusChecksPending (not .HasPendingPullRequestMerge)}}
+											<input type="hidden" name="merge_when_checks_succeed" value="1"/>
+										{{end}}
+										<div class="field">
+											<input type="text" name="merge_title_field" value="{{.Issue.PullRequest.GetDefaultMergeMessage}}">
+										</div>
+										<div class="field">
+											<textarea name="merge_message_field" rows="5" placeholder="{{$.i18n.Tr "repo.editor.commit_message_desc"}}">Reviewed-on: {{$.Issue.HTMLURL}}&#13;&#10;{{$approvers}}</textarea>
+										</div>
+										<button class="ui {{$mergeButtonColor}} button" type="submit" name="do" value="merge">
```

Comment:
What about declaring this as an HTML `<template>` and inserting it dynamically with JS?
That way you can easily ensure that it exists only once.
Of course, the `delete branch` checkbox could also be a part of this.
Alternative: Extract into a separate template that you then call.